### PR TITLE
Fixes #18490 - run eslint on all files in webpack

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

**Comment by gailsteiger** [comment]
what other files other than .js do you have in mind?

**Comment by tbrisker** [comment]
@gailsteiger for some reason not all files are matched by the existing pattern, even though they should.

**Comment by gailsteiger** [comment]
Wow. Which files is it missing? Nice catch.

Does ` webpack/assets/**/*/*.js` work?

**Comment by tbrisker** [comment]
@gailsteiger I didn't manage to figure out the exact condition, but when testing #4280 a lot of the files weren't being linted by `npm run lint`, which led to a lot of barking from the hound.

**Comment by tbrisker** [comment]
@gailsteiger Actually I know why - bash was interpolating the `*`'s so only the top level was being linted, nothing under the other folders.

File: package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "select2": "~3.5.2-browserify"
   },
   "scripts": {
-    "lint": "./node_modules/.bin/eslint -c .eslintrc webpack/assets/**/*.js || exit 0",
+    "lint": "./node_modules/.bin/eslint -c .eslintrc webpack/assets/**/* || exit 0",
```
Comment (ohadlevy) [inline package.json]: this will catch also css files and fail.

i believe ```webpack/**/*/*js``` works for me.

Comment (tbrisker) [reply package.json]: no, eslint only checks .js files unless --ext is used.
Also that does not work, you should see a warning on API.js.

**Comment by tbrisker** [comment]
Amended to remove the glob, eslint can be passed an entire directory as a target.

**Pull request merged**